hs_collection_resource/tests/test_collection.py

```diff
@@ -0,0 +1,349 @@
+        resp_json = json.loads(response.content)
+        self.assertEqual(resp_json["status"], "success")
+        self.assertEqual(resp_json["user_permission"], "Edit")
+        self.assertEqual(resp_json["current_sharing_status"], "Private")
+        self.assertEqual(resp_json["new_sharing_status"], "")
+        self.assertEqual(resp_json["metadata_status"], "Sufficient to make public")
+        self.assertEqual(self.resCollection.metadata.collection.resources.all().count(), 2)
+
+        # make User 2's Gen4 discoverable
+        self.resGen4.raccess.discoverable = True
+        self.resGen4.raccess.save()
+
+        # User 2 adds his dicoverable res 4
+        response = self.api_client.post(url_to_update_collection, {'resource_id_list': \
+                        [self.resGen1.short_id, self.resGen2.short_id, self.resGen4.short_id]})
+        resp_json = json.loads(response.content)
+        self.assertEqual(resp_json["status"], "success")
+        self.assertEqual(resp_json["user_permission"], "Edit")
+        self.assertEqual(resp_json["current_sharing_status"], "Private")
+        self.assertEqual(resp_json["new_sharing_status"], "")
+        self.assertEqual(resp_json["metadata_status"], "Sufficient to make public")
+        self.assertEqual(self.resCollection.metadata.collection.resources.all().count(), 3)
+
+    def test_custom_logic_in_hs_core_1(self):
+
+        # this func tests that the public collection should downgrade to private if its any member resource is changed to private
+
+        self.api_client.logout()
+        # user 2 login
+        self.api_client.login(username='user2', password='mypassword2')
+
+        ##### test custom logic for collection in hs_core
+        # check gen res 5 current sharing status --> private
+        self.assertEqual(self.resGen5_user2.raccess.public, False)
+        self.assertEqual(self.resGen5_user2.raccess.discoverable, False)
+
+        # make gen res 5 public (not working because it does not meet public precondition: no resouece file in generic res)
+        # url_to_set_resource_flag_for_resGen5 = self.url_to_set_resource_flag.format(self.resGen5_user2.short_id)
+        # response = self.api_client.post(url_to_set_resource_flag_for_resGen5, {'t': 'make_public'}, HTTP_REFERER='http://foo/bar')
+
+        # make gen res 5 public manaully
+        self.resGen5_user2.raccess.public = True
+        self.resGen5_user2.raccess.discoverable = True
+        self.resGen5_user2.raccess.save()
+
+        # check gen res 5 current sharing status --> public
+        self.assertEqual(self.resGen5_user2.raccess.public, True)
+        self.assertEqual(self.resGen5_user2.raccess.discoverable, True)
+
+        # check collection current sharing status --> private
+        self.assertEqual(self.resCollection2_user2.raccess.public, False)
+        self.assertEqual(self.resCollection2_user2.raccess.discoverable, False)
+
```

Comment:
Here you should test that the collection can't be made public before adding a resource to the collection resource. Use the resource property ```can_be_public_or_discoverable```